docs/content/python/getting-started/application/filter-data.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Filter collections of data
+description: |
+  Many applications perform authorization over large collections of data
+  that cannot be loaded into memory. Often index pages showing users a
+  number of resources, like the repositories they can access, will need to
+  use data filtering. The data filtering API provides support for these
+  use cases.
+weight: 4
+---
+
+# Filter collections of data
+
+Many applications perform authorization over large collections of data
+that cannot be loaded into memory. Often index pages showing users a
+number of resources, like the repositories they can access, will need to
+use data filtering. The data filtering API provides support for these
+use cases.
+
+## Authorizing queries
+
+In [Add authorization enforcement](enforce) we discussed enforcing
+*resource-level authorization*. The `authorize` method tells you whether
+*a specific resource* is authorized. But what about endpoints that need
+to authorize a collection of resources?
+
+You can authorize queries to obtain a set of resources that are
+authorized for a particular user and action. Let's go back to our
+repository example, but this time we are listing repositories the user
+has access to:
+
+{{< literalInclude
+    path="examples/add-to-your-application/python/app/data_filtering.py"
+    from="docs: begin-list-route"
+    to="docs: end-list-route"
+    >}}
+
+To use this API, you must pass some additional information to
+`register_class` so that Oso knows how to query for your application's
+objects.
+
+TODO make sure this matches data filtering guides
+
+## Implementing data filtering query functions
+
+To use data filtering, you tell Oso how to query for the objects in your
```

Comment:
You need to be really, really clear what "query" you are referring to throughout.

Reply:
oof yeah